src/contentmanager.cpp

```diff
@@ -539,11 +540,13 @@ void ContentManager::updateLibrary() {
 
 #define CATALOG_URL "library.kiwix.org"
 void ContentManager::updateRemoteLibrary(const QString& content) {
-    m_remoteLibrary = kiwix::Library();
-    kiwix::Manager manager(&m_remoteLibrary);
-    manager.readOpds(content.toStdString(), CATALOG_URL);
-    emit(this->booksChanged());
-    emit(this->pendingRequest(false));
+    QtConcurrent::run([=]() {
+        m_remoteLibrary = kiwix::Library();
```

Comment:
What happens if someone is using the remote Library while this function is running in a different thread ?
We should probably but a lock somewhere.